### PR TITLE
Revert `fillBytes` method to `bigIntToBytes` for lower golang version

### PR DESCRIPTION
**Pull request by ahangsu**
# Summary

We are probably not going to higher golang version (1.15+) at this point, while `fillBytes` from `big.Int` is introduced in 1.15.
Changing all the `fillBytes` in `ABI` to self-made `bigIntToBytes` in favor of lower go version.

### CODE REVIEW

**Review by jannotti — DISMISSED** [review]
I might have preferred a fill-in method with the exact same signature and semantics as FillBytes, so the change would be even smaller, but I'mm totally fine with this.

Let's merge after checks pass.

File: data/abi/abi_encode.go [hunk 1]
```diff
@@ -25,6 +25,25 @@ import (
...
+	}
+	if uint(x.BitLen()) > byteLen*8 {
+		return nil, fmt.Errorf("ABI: big Int To Bytes error: integer byte length > given byte length")
+	}
+
+	buffer := make([]byte, byteLen)
+	intBytes := x.Bytes()
+	for i := 0; i < len(intBytes); i++ {
+		buffer[len(buffer)-1-i] = intBytes[len(intBytes)-1-i]
+	}
```
Comment (tsachiherman) [inline data/abi/abi_encode.go:43]: I could be reading this wrong, but it looks to me like you're trying to do
```golang
copy(buffer[byteLen-len(intBytes):], intBytes)
```

Comment (jannotti) [inline data/abi/abi_encode.go:43]: Could this have been `copy(buffer, x.Bytes())`?

Comment (ahangsu) [reply data/abi/abi_encode.go:43]: yes exactly, I am somewhat rusty on golang now, will follow JJ's comments and change this line accordingly.

**Comment by jannotti** [comment]
> I might have preferred a fill-in method with the exact same signature and semantics as FillBytes, so the change would be even smaller, but I'mm totally fine with this.
> 
> Let's merge after checks pass.

For what it's worth, emulating FillBytes exactly would be annoying, as it is willing to take a buffer of any size that's big enough.  Our use case only needs to work on exactly the right size buffer.

File: data/abi/abi_encode.go (hunk 1 above)
Comment (ahangsu) [reply data/abi/abi_encode.go:43]: yes, please see Tsachi's comment above.

File: data/abi/abi_json_test.go [hunk 2]
```diff
@@ -31,14 +31,18 @@ func TestRandomAddressEquality(t *testing.T) {
...
 		randomAddrInt, err := rand.Int(rand.Reader, upperLimit)
 		require.NoError(t, err, "cryptographic random int init fail")
 
-		randomAddrInt.FillBytes(addrBasics[:])
-		randomAddrInt.FillBytes(addrABI)
+		expected, err := bigIntToBytes(randomAddrInt, uint(addressByteSize))
+		require.NoError(t, err, "big int to byte conversion error")
+		for i := 0; i < addressByteSize; i++ {
+			addrBasics[i] = expected[i]
+			addrABI[i] = expected[i]
```
Comment (jasonpaulos) [inline data/abi/abi_json_test.go:44]: `copy` could be used here as well instead of this loop

**Review by jasonpaulos — APPROVED** [review]
Looks correct. I agree with the earlier comments about `copy` and I left another, but they are minor

**Comment by ahangsu** [comment]
I tried to emulate a method like FillBytes before, but we would need to introduce new type alias for big Int (in favor of defining new methods), and a bunch of type casting to use new method, and I am reluctant to create a larger delta.

The current fix should be sufficient for the purpose of casting big.Int to specific length of byte slice, please let me know your thoughts.

**Review by jasonpaulos — APPROVED** [review]

File: data/abi/abi_encode.go [hunk 3]
```diff
@@ -25,6 +25,23 @@ import (
 	"strings"
 )
 
+// bigIntToBytes casts non-negative big integer to byte slice with specific byte length
+// NOTE: THIS IS A WORKAROUND FOR `fillBytes` METHOD BEFORE GOLANG 1.15+
+//       SHOULD BE REMOVED AFTER WE MOVE TO HIGHER VERSION
```
Comment (winder) [inline data/abi/abi_encode.go:30]: nit: putting deprecated in the comment here triggers some IDEs (like goland) to cross out the method.
```suggestion
// bigIntToBytes casts non-negative big integer to byte slice with specific byte length
// DEPRECATED: THIS IS A WORKAROUND FOR `fillBytes` METHOD BEFORE GOLANG 1.15+
//       SHOULD BE REMOVED AFTER WE MOVE TO HIGHER VERSION
```

File: data/abi/abi_encode_test.go [hunk 4]
```diff
@@ -1162,7 +1165,7 @@ func TestParseArgJSONtoByteSlice(t *testing.T) {
 
 	for i, test := range tests {
 		t.Run(fmt.Sprintf("index=%d", i), func(t *testing.T) {
-			applicationArgs := [][]byte{}
+			applicationArgs := make([][]byte, 0)
```
Comment (jannotti) [inline data/abi/abi_encode_test.go:1168]: Need not put a 0 here.

**Review by jannotti — APPROVED** [review]

**Pull request merged**